sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java

```diff
@@ -593,10 +593,15 @@ static String getDatasetLocation(
 
   static void verifyTablePresence(DatasetService datasetService, TableReference table) {
     try {
-      datasetService.getTable(table);
+      Table fetchedTable = datasetService.getTable(table);
+      if (fetchedTable == null) {
+        throw new IOException("Table does not exist.");
```

Comment:
can we print the table name as well? or may be add some actionable steps for users to take like create the table first with the right schema?

Reply:
ah note that this exception is caught again on line 603 and formatted properly. Why do we do this? because `datasetService.getTable` can also throw an exception so we deal with both cases there